sajomathews/codeinsight
# Feature/integrate langchain

## Description

**Purpose:**
This PR introduces a pytest test case to verify that `LlamaCpp` Large Language Model (LLM) works seamlessly with Langchain.

**Changes Made:**
- Created a new test file `test_llama_cpp.py` in the `tests` directory.
- Added a test case `test_llama_cpp_with_langchain` that utilizes `LlamaCpp` to generate some code using example Python code.
- The test asserts that the code generated is not empty, verifying proper interaction between `LlamaCpp` and Langchain.

**Testing:**
- Tested the new test case locally using pytest to ensure it runs successfully.
- Verified that `LlamaCpp` and Langchain are properly set up in the development environment.
- Confirmed that the test case accurately checks the code summarization functionality.

## How to Test

To test `LlamaCpp` and Langchain, follow these steps after applying this PR:

1. Ensure you have the necessary dependencies (including `LlamaCpp` and Langchain) set up in your development environment.
2. Ensure a codellama model is available at the path referred to by the test
3. Run the pytest tests by executing the following command in the project's root directory:

   ```bash
   poetry run pytest
